src/App.tsx

```diff
@@ -1,19 +1,144 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
+import ClassNames from 'classnames';
+import { User, TodoWidthUser } from './react-app-env';
 
+import todos from './api/todos';
 import users from './api/users';
 
-const App: React.FC = () => {
+import { TodoList } from './components/TodoList/TodoList';
+import './App.scss';
+
+const getUsersById = (userId: number): User | null => {
+  return users.find(user => user.id === userId) || null;
+};
+
+const todosWidthUser: TodoWidthUser[] = todos.map((todo) => ({
+  ...todo,
+  user: getUsersById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [newTodoWidthUser, setNewTodoWidthUser] = (
+    useState<TodoWidthUser[]>([...todosWidthUser])
+  );
+  // const [newIdTodo, setNewIdTodo] = useState([...todos].reverse()[0].id); // for line 34
+  const [newTitleName, setNewTitleName] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [newUserId, setNewUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+  const [newStatusTodo, setNewStatusTodo] = useState(false);
+
+  const addNewTodo = () => {
+    const newTodo: TodoWidthUser = {
+      userId: newUserId,
+      // id: Number(setNewIdTodo(newIdTodo + 1)), === NaN // ???
+      id: Date.now(),
+      title: newTitleName,
+      completed: newStatusTodo,
+      user: getUsersById(newUserId),
+    };
+
+    setNewTodoWidthUser((currentTodo) => [...currentTodo, newTodo]);
+  };
+
+  const handleSubmitForm = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setHasTitleError(!newTitleName);
+    setHasUserIdError(!newUserId);
+
+    if (newTitleName && newUserId) {
+      addNewTodo();
+      setNewTitleName('');
+      setNewUserId(0);
+    }
+  };
+
+  const addErrorClass = (
+    defaultClass: string,
+    checkedValue: boolean,
+  ) => (
+    ClassNames(defaultClass, {
+      error: checkedValue,
+    })
+  );
+
   return (
     <div className="App">
-      <h1>Add todo form</h1>
+      <h1 className="title">Add TODO form !</h1>
+
+      <form
+        className="box form"
+        onSubmit={handleSubmitForm}
+      >
+        <div
+          className={addErrorClass('select is-info', hasUserIdError)}
+        >
+          <select
+            className="selMain"
+            name="userName"
+            value={newUserId}
+            onChange={(event) => {
```

Comment:
You can move this logic with callback before render. This optimizes the code in the layout.